### DISCUSSION TITLE
Can optional uint32 value be 0

### DISCUSSION THREAD

**Discussion by lemueldsouza** [protobuf-c/protobuf-c#661] [Q&A] [answered]
This is just a sample message 
Message Data{
    optional uint64 userid = 1;
    repeated string username = 2;  
    optional uint32 status = 3;
}

Case 1
If status = non-zero value
When I use the data__pack(const Data \* message ,uint8_t \* out) function to pack the data, the size of 'out' is 100, and the function returns 100.

Case 2
If status = 0 (for same values of userid and username as in Case 1)
When I use the data__pack(const Data \* message ,uint8_t \* out) function to pack the data, the size of 'out' is 99 but the function still returns 100.

If you notice the 2 cases above, if a value of 0 is used for status, the size of the output buffer is always 1 less than the size if status is a non-zero value. Is this a protobuf bug?

**Comment by edmonds** [comment, marked as answer]
Hello,

This is expected behavior. Protocol Buffers uses a variable-width encoding for integers. See the [Encoding](https://developers.google.com/protocol-buffers/docs/encoding) section in the Protocol Buffers developer guide.

**Comment by lemueldsouza** [comment]
Ok, but wouldn't this be an issue if I were to send this data over HTTP?

Content-Length: 100
Content-Type: application/protobuf

<Body containing the protobuf encoded data - 99>

The receiver would now continue to wait until it receives Content-Length of 100 and would eventually time out since it receives only 99.

**Comment by edsiper** [comment]
Http client or server side just know about bytes, if the client set a
content length of 100 but only send 99... the client have a serious bug.
